codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperFuelWorker.res

```diff
@@ -3,80 +3,97 @@ open Belt
+let getEventTag = (eventConfig: Types.fuelEventConfig) => {
+  switch eventConfig.kind {
+  | Mint => mintEventTag
+  | LogData({logId}) => logId
+  }
 }
 
-let makeGetRecieptsSelectionOrThrow = (~contracts: array<contract>) => {
+let makeWorkerConfigOrThrow = (~contracts: array<Types.fuelContractConfig>, ~chain) => {
+  let eventRouter = EventRouter.empty()
+  let contractByEvent = Utils.WeakMap.make()
+  let nonWildcardLogDataRbsByContract = Js.Dict.empty()
+  let wildcardLogDataRbs = []
+  let contractsWithMint = Utils.Set.make()
+  let hasWildcardMint = ref(false)
+
+  contracts->Belt.Array.forEach(contract => {
+    let nonWildcardLogDataRbs = []
+    contract.events->Array.forEach(eventConfig => {
+      eventRouter->EventRouter.addOrThrow(
+        getEventTag(eventConfig),
+        eventConfig,
+        ~contractName=contract.name,
+        ~eventName=eventConfig.name,
+        ~chain,
+        ~isWildcard=eventConfig.isWildcard,
+      )
+      contractByEvent->Utils.WeakMap.set(eventConfig, contract)->ignore
+
+      switch eventConfig {
+      | {kind: Mint, isWildcard: true} => hasWildcardMint := true
+      | {kind: Mint} => contractsWithMint->Utils.Set.add(contract.name)->ignore
+      | {kind: LogData({logId}), isWildcard} => {
+          let rb = logId->BigInt.fromStringUnsafe
+          if isWildcard {
+            wildcardLogDataRbs->Array.push(rb)->ignore
+          } else {
+            nonWildcardLogDataRbs->Array.push(rb)->ignore
+          }
+        }
+      }
+    })
+    nonWildcardLogDataRbsByContract->Js.Dict.set(contract.name, nonWildcardLogDataRbs)
+  })
+
+  {
+    eventRouter,
+    contractByEvent,
+    wildcardLogDataRbs,
+    nonWildcardLogDataRbsByContract,
+    contractsWithMint,
+    hasWildcardMint: hasWildcardMint.contents,
+  }
+}
+
+let makeGetRecieptsSelection = (
+  ~wildcardLogDataRbs,
+  ~nonWildcardLogDataRbsByContract,
+  ~contractsWithMint,
+  ~hasWildcardMint,
+  ~contracts: array<Types.fuelContractConfig>,
+) => {
   let logDataReceiptTypeSelection: array<Fuel.receiptType> = [LogData]
   let mintReceiptTypeSelection: array<Fuel.receiptType> = [Mint]
 
   // only transactions with status 1 (success)
   let txStatusSelection = [1]
 
-  let nonWildcardRbsByContract = Js.Dict.empty()
-  let contractsWithMint = Utils.Set.make()
-  let wildcardMint = ref(false)
-  let wildcardRbs = []
-
   let wildcardMintSelection: HyperFuelClient.QueryTypes.receiptSelection = {
     receiptType: mintReceiptTypeSelection,
     txStatus: txStatusSelection,
   }
 
-  contracts->Array.forEach(contract => {
-    let nonWildcardRbs = []
-    contract.events->Array.forEach(event => {
-      switch event {
-      | {mint: true, logId: _} =>
-        Js.Exn.raiseError(`Mint event ${event.name} is not allowed to have a logId`)
-      | {mint: true, isWildcard: true} =>
-        if contractsWithMint->Utils.Set.size !== 0 {
-          Js.Exn.raiseError(`Wildcard Mint event is not allowed together with non-wildcard Mint events`)
```

Comment:
Now this errors handled by EventRouter. Also moved all the data prep to the makeWorkerConfigOrThrow funciton

Reply:
Amazing, it's much easier to read and understand now 👍🏼